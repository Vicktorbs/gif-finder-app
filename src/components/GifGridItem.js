import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {
    return (
        <a href={ url } target='_blank' rel='noreferrer' className="card card-background">
            <div className="animate__animated animate__fadeIn">
                <img src={ url } alt={ title } />
                <p>{ title }</p>
            </div>
        </a>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}