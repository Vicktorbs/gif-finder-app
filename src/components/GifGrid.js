import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    // key: sgatASouL6QA0OloiTwxS5gAVHGXDGET
    // const [images, setImages] = useState([])
    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <li className="cards-container" key={ category }>
                <h3 className="animate__animated animate__fadeIn card-title">{ category }</h3>
                { loading && <p>Loading...</p> }
                <div className="card-grid">
                    { 
                        images.map(image => 
                            <GifGridItem { ...image } key={ image.id } />
                        )
                    }
                </div>
            </li>
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}