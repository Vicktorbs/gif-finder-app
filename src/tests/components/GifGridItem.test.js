import { shallow } from 'enzyme';
import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';

const title = "i just love her naruto shippuden GIF"
const url = "https://media1.giphy.com/media/qKr1nIRfIQPYI/giphy.gif?cid=c4f74b61bdxuxtgjry3au947bwjp4fc6uncmdmfvt0b48g7v&rid=giphy.gif&ct=g"
const wraper = shallow(<GifGridItem key="123" title={ title } url={ url } />) 

describe('Test on file <GifGridItem />', () => {

    test('should show GifGridItem correctly', () => {
        expect(wraper).toMatchSnapshot()
    })

    test('should has a pharagrap with the title', () => {
        const p = wraper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    
    test('<img> should has the same url and alt from props', () => {
        const img = wraper.find('img');
        // .props extrae solo las propiedades de el elemento seleccionado por el wrapper
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('<img> should has the class animate__fadeIn', () => {
        const div = wraper.find('div');
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true)
    })
    

})
