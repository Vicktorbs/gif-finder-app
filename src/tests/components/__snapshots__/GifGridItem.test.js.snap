// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on file <GifGridItem /> should show GifGridItem correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="i just love her naruto shippuden GIF"
    src="https://media1.giphy.com/media/qKr1nIRfIQPYI/giphy.gif?cid=c4f74b61bdxuxtgjry3au947bwjp4fc6uncmdmfvt0b48g7v&rid=giphy.gif&ct=g"
  />
  <p>
    i just love her naruto shippuden GIF
  </p>
</div>
`;
